documentation/src/main/asciidoc/chinese/chapter-8-值的映射.asciidoc

```diff
@@ -0,0 +1,268 @@
+4. 不同点：考虑到1. 和3. ，绝不会有未映射的值。
+
+*`String` to enum*
+
+1. 相同点: 如果没有定义一个显式映射，target enum会由与其常量同名的source String映射得来。
+2. 相同点: `<ANY_UNMAPPED>` 同样仅会仅处理显式映射，并把其余值都处理为默认值。
+3. 相同点: `<ANY_REMAINING>` 会为target enum中的每个值创造一个映射，并把未匹配的值处理为switch语句中的默认值。
+4. 不同点: 需要设定一个switch/default值来保证一个确定性的结果（枚举的值是有限的集合，String有无穷的可能）。如果 `<ANY_REMAINING>` 和 `<ANY_UNMAPPED`>  都没有指定，那么会有一个warning。
+
+=== 自定义名称映射
+
+当没有定义 `@ValueMapping` 时，source enum中的每个枚举都会映射为常量名称相同的target enum。
+
+然而，在一些场景中，source enum需要先做一些转换再做映射，比如source enum要加个尾缀再转换为target enum。
+
+.Enum types
+====
+[source, java, linenums]
+[subs="verbatim,attributes"]
+----
+public enum CheeseType {
+
+    BRIE,
+    ROQUEFORT
+}
+
+public enum CheeseTypeSuffixed {
+
+    BRIE_TYPE,
+    ROQUEFORT_TYPE
+}
+----
+====
+
+.Enum mapping method with custom name transformation strategy
+====
+[source, java, linenums]
+[subs="verbatim,attributes"]
+----
+@Mapper
+public interface CheeseMapper {
+
+    CheeseMapper INSTANCE = Mappers.getMapper( CheeseMapper.class );
+
+    @EnumMapping(nameTransformationStrategy = "suffix", configuration = "_TYPE")
+    CheeseTypeSuffixed map(CheeseType cheese);
+
+    @InheritInverseConfiguration
+    CheeseType map(CheeseTypeSuffix cheese);
+}
+----
+====
+
+.Enum mapping method with custom name transformation strategy result
+====
+[source, java, linenums]
+[subs="verbatim,attributes"]
+----
+// GENERATED CODE
+public class CheeseSuffixMapperImpl implements CheeseSuffixMapper {
+
+    @Override
+    public CheeseTypeSuffixed map(CheeseType cheese) {
+        if ( cheese == null ) {
+            return null;
+        }
+
+        CheeseTypeSuffixed cheeseTypeSuffixed;
+
+        switch ( cheese ) {
+            case BRIE: cheeseTypeSuffixed = CheeseTypeSuffixed.BRIE_TYPE;
+            break;
+            case ROQUEFORT: cheeseTypeSuffixed = CheeseTypeSuffixed.ROQUEFORT_TYPE;
+            break;
+            default: throw new IllegalArgumentException( "Unexpected enum constant: " + cheese );
+        }
+
+        return cheeseTypeSuffixed;
+    }
+
+    @Override
+    public CheeseType map(CheeseTypeSuffixed cheese) {
+        if ( cheese == null ) {
+            return null;
+        }
+
+        CheeseType cheeseType;
+
+        switch ( cheese ) {
+            case BRIE_TYPE: cheeseType = CheeseType.BRIE;
+            break;
+            case ROQUEFORT_TYPE: cheeseType = CheeseType.ROQUEFORT;
+            break;
+            default: throw new IllegalArgumentException( "Unexpected enum constant: " + cheese );
+        }
+
+        return cheeseType;
+    }
+}
+----
+====
+
+MapStruct提供了如下开箱即用的名称转换策略：
+
+* _suffix_ - 给source enum加尾缀
+* _stripSuffix_ - 给source enum加strip后的尾缀
+* _prefix_ - 给source enum加前缀
+* _stripPrefix_ - 给source enum加strip后的前缀
+* _case_ - 给source enum提供各种样式的名称转换，提供的样式有：
+** _upper_ - 将source enum转换为大写样式
+** _lower_ - 将source enum转换为大写样式
+** _capital_ - 把source enum中每个单词的第一个字母大写，其余字母小写（单词以"_"分割）
+
+也支持注册自定义策略，更多信息请参见 <<custom-enum-transformation-strategy>> 
```

Comment:
Would you be so kind and add the new section: 

https://github.com/mapstruct/mapstruct/commit/6a394ad466b5bda8b17e2406b56be34be3312e21#diff-aaca9b52e434c4fb6940e4405327fe73da5f3438442109c4c8c6f858c0d6ddee